doc/classes/TileMapLayer.xml

```diff
@@ -0,0 +1,309 @@
+			<return type="Vector2i[]" />
+			<description>
+				Returns a [Vector2i] array with the positions of all cells containing a tile. A cell is considered empty if its source identifier equals [code]-1[/code], its atlas coordinates identifiers is [code]Vector2(-1, -1)[/code] and its alternative identifier is [code]-1[/code].
+			</description>
+		</method>
+		<method name="get_used_cells_by_id" qualifiers="const">
+			<return type="Vector2i[]" />
+			<param index="0" name="source_id" type="int" default="-1" />
+			<param index="1" name="atlas_coords" type="Vector2i" default="Vector2i(-1, -1)" />
+			<param index="2" name="alternative_tile" type="int" default="-1" />
+			<description>
+				Returns a [Vector2i] array with the positions of all cells containing a tile. Tiles may be filtered according to their source ([param source_id]), their atlas coordinates ([param atlas_coords]) or alternative id ([param alternative_tile]).
+				If a parameter has its value set to the default one, this parameter is not used to filter a cell. Thus, if all parameters have their respective default value, this method returns the same result as [method get_used_cells].
+				A cell is considered empty if its source identifier equals [code]-1[/code], its atlas coordinates identifiers is [code]Vector2(-1, -1)[/code] and its alternative identifier is [code]-1[/code].
+			</description>
+		</method>
+		<method name="get_used_rect" qualifiers="const">
+			<return type="Rect2i" />
+			<description>
+				Returns a rectangle enclosing the used (non-empty) tiles of the map.
+			</description>
+		</method>
+		<method name="has_body_rid" qualifiers="const">
+			<return type="bool" />
+			<param index="0" name="body" type="RID" />
+			<description>
+				Returns whether the provided [param body] [RID] belongs to one of this [TileMapLayer]'s cells.
+			</description>
+		</method>
+		<method name="local_to_map" qualifiers="const">
+			<return type="Vector2i" />
+			<param index="0" name="local_position" type="Vector2" />
+			<description>
+				Returns the map coordinates of the cell containing the given [param local_position]. If [param local_position] is in global coordinates, consider using [method Node2D.to_local] before passing it to this method. See also [method map_to_local].
+			</description>
+		</method>
+		<method name="map_pattern">
+			<return type="Vector2i" />
+			<param index="0" name="position_in_tilemap" type="Vector2i" />
+			<param index="1" name="coords_in_pattern" type="Vector2i" />
+			<param index="2" name="pattern" type="TileMapPattern" />
+			<description>
+				Returns for the given coordinate [param coords_in_pattern] in a [TileMapPattern] the corresponding cell coordinates if the pattern was pasted at the [param position_in_tilemap] coordinates (see [method set_pattern]). This mapping is required as in half-offset tile shapes, the mapping might not work by calculating [code]position_in_tile_map + coords_in_pattern[/code].
```

Comment:
Questioning if this method should even be public, or if there's another way to achieve what this method needs to be used for in a way that is more accessible and less "verbose".

Reply:
I think it's fine and useful enough in some cases. It was already present in TileMap so I don't want to remove it now.